openqaoa/workflows/optimizer.py

```diff
@@ -430,42 +536,206 @@ def compile(self, problem: QUBO = None, verbose: bool = True):
             !NotYetImplemented! Set true to have a summary of RQAOA to displayed after compilation
         """
 
-        self.qaoa.compile(problem, verbose=False)
+        # save the original problem
+        self.problem = problem 
+
+        # Create the qaoa object with the properties
+        self._q = QAOA(self.device)
+        self._q.circuit_properties  = self.circuit_properties
+        self._q.backend_properties  = self.backend_properties
+        self._q.classical_optimizer = self.classical_optimizer
 
-    def optimize(self):
+        # compile qaoa object
+        self._q.compile(problem, verbose=verbose)
+
+
+        if self.rqaoa_parameters.rqaoa_type.lower() == "custom":
+
+            n_cutoff = self.rqaoa_parameters.n_cutoff
```

Comment:
I wouldn't introduce new variables but use the self. @vishal-ph ?

Reply:
I added them to make the code more readable but if it's not optimal I'll use the self. variables directly

Reply:
Yeah, I know. I think I prefer shorter scripts. Let's leave it like this for the moment  then :)